### PR TITLE
feat(ci): test wixext & build msis

### PR DESCRIPTION
**Pull request by amdprophet**
Signed-off-by: Justin Kolberg <amd.prophet@gmail.com>

Adds MSI building to the dev & release CI workflows & adds a test job for the C# WixExt to the PR workflow.

### CODE REVIEW

File: .github/workflows/release_builds.yml [hunk 1]
```diff
@@ -390,6 +410,50 @@ jobs:
...
+          name: windows_amd64
+          path: ./otelcolbuilder/cmd
+
+      - name: Rename binary for windows/amd64
+        working-directory: ./otelcolbuilder/cmd
+        run: mv otelcol-sumo-*-sumo-*-windows_amd64.exe otelcol-sumo-windows_amd64.exe
+
+      - name: DEBUG 1
+        run: ls -l ./otelcolbuilder/cmd
+
```
Comment (swiatekm-sumo) [inline .github/workflows/release_builds.yml:439]:
```suggestion
```

File: ci/get_productversion.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+# ProductVersion.
+#
+# https://learn.microsoft.com/en-us/windows/win32/msi/productversion
+# MAJOR.MINOR.BUILD.INTERNAL
+
+declare -i major_version
+declare -i minor_version
+declare -i build_version
+declare -i internal_version
+#declare channel
```
Comment (swiatekm-sumo) [inline ci/get_productversion.sh:15]:
```suggestion
```

File: ci/get_productversion.sh [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    ot_channel_version="${BASH_REMATCH[7]}"
+    sumo_channel="${BASH_REMATCH[9]}"
+    sumo_channel_version="${BASH_REMATCH[10]}"
+else
+    echo "Error: Tag can not be converted to a Windows ProductVersion: ${version_tag}" >&2
+    exit 1
+fi
+
+if [[ $ot_channel == "sumo" ]]; then
+    #channel="${sumo_channel}"
```
Comment (swiatekm-sumo) [inline ci/get_productversion.sh:38]:
```suggestion
```

File: ci/get_productversion.sh [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    if [[ $sumo_channel != "" ]]; then
+        internal_version="${sumo_channel_version}"
+    else
+        internal_version="${ot_channel_version}"
+    fi
+elif [[ $ot_channel != "" ]]; then
+    #channel="${ot_channel}"
```
Comment (swiatekm-sumo) [inline ci/get_productversion.sh:45]:
```suggestion
```

**Review by swiatekm-sumo — COMMENTED** [review]
Found some debugging leftovers. The logic looks sound, though I haven't tested the GHA jobs manually.

**Review by swiatekm-sumo — APPROVED** [review]

**Review by perk-sumo — APPROVED** [review]
🚀

**Pull request merged**